### PR TITLE
Datahub: Add tabs to homepage and routing

### PR DESCRIPTION
**Pull request by tkohr**
The PR re-organizes the datahub `search-page` into a `home-page`, introducing tabs to a `news-page` and an `organisations-page`. The new routes are added from the datahub via a `datahub-router-init-service` without touching on the router lib.
Currently, the PR touches on the `RouterConfigModel`, the `RouterEffects` and the `SearchRouteReuseStrategy` of the lib in the last commit, however.

TODOs:

- [x] keep latest search state when switching tabs
- [x] fix header layout
- [x] redirect home
- [x] prevent touching on lib
- [x] tests
- [x] fix infiniteScroll after navigating to md record

tabs

### CODE REVIEW

**Review by fgravin — COMMENTED** [review]
Looking good so far.
The infinite scroll breaks if you change the tab and then go back on the `datasets` tab.
Looks like the sticky header is less performant than before too.

This morning the search was reset when going back to datasets, what did you change ?
Thanks

File: apps/datahub/src/app/home/home-header/home-header.component.html [hunk 1]
```diff
@@ -29,21 +29,42 @@
...
+        datahub.header.datasets
+      </div>
+      <div
+        class="uppercase truncate text-white hover:text-primary-lighter hover:underline cursor-pointer"
+        [class]="
+          (currentRoute$ | async) === 'organisations'
+            ? 'text-primary'
+            : 'text-white'
+        "
+        (click)="onOrganisationsClick()"
```
[the remark this answers is not in the document]
Comment (tkohr) [reply apps/datahub/src/app/home/home-header/home-header.component.html:63]: Indeed, I kept the `consts` for less hard coding, though.

File: apps/datahub/src/app/home/home-header/home-header.component.html [hunk 2]
```diff
@@ -29,21 +29,42 @@
...
+        "
         (click)="onDatasetsClick()"
         translate=""
       >
-        datasets
+        datahub.header.datasets
+      </div>
+      <div
+        class="uppercase truncate text-white hover:text-primary-lighter hover:underline cursor-pointer"
+        [class]="
```
[the remark this answers is not in the document]
Comment (tkohr) [reply apps/datahub/src/app/home/home-header/home-header.component.html:58]: Oh yes, great simplification, thanks!

File: apps/datahub/src/app/home/home-header/home-header.component.html [hunk 3]
```diff
@@ -35,32 +35,25 @@
     >
       <div
         class="uppercase truncate text-white hover:text-primary-lighter hover:underline cursor-pointer"
-        [class]="
-          (currentRoute$ | async) === 'news' ? 'text-primary' : 'text-white'
-        "
-        (click)="onNewsClick()"
+        routerLinkActive="text-primary"
+        routerLink="{{ ROUTE_NEWS }}"
```
Comment (fgravin) [inline apps/datahub/src/app/home/home-header/home-header.component.html:39]:
```suggestion
        [routerLink]="ROUTE_NEWS"
```

**Comment by tkohr** [comment]
> Looking good so far. The infinite scroll breaks if you change the tab and then go back on the `datasets` tab. Looks like the sticky header is less performant than before too.

Thanks for the review and suggested improvements! Indeed, the infinite scroll breaks, when navigating to a metadata record and back (not in between tabs though). Still have to figure this one out.

> This morning the search was reset when going back to datasets, what did you change ? Thanks

This commit https://github.com/geonetwork/geonetwork-ui/pull/280/commits/18d56f0869b466a56fd21f0886f3c99c2b4643b4 prevents resetting the search and updates it with the latest value (which is more consistent with search input value).

**Comment by fgravin** [comment]
> > Looking good so far. The infinite scroll breaks if you change the tab and then go back on the `datasets` tab. Looks like the sticky header is less performant than before too.
> 
> Thanks for the review and suggested improvements! Indeed, the infinite scroll breaks, when navigating to a metadata record and back (not in between tabs though). Still have to figure this one out.
> 
> Didn't notice any lags with the sticky header yet.
> 
> > This morning the search was reset when going back to datasets, what did you change ? Thanks
> 
> This commit [18d56f0](https://github.com/geonetwork/geonetwork-ui/commit/18d56f0869b466a56fd21f0886f3c99c2b4643b4) prevents resetting the search and updates it with the latest value (which is more consistent with search input value).

Hmm ok, so you are actually requesting a search.
Fine for now, remember it should not be necessary if the component remains

**Comment by tkohr** [comment]
> > This commit [18d56f0](https://github.com/geonetwork/geonetwork-ui/commit/18d56f0869b466a56fd21f0886f3c99c2b4643b4) prevents resetting the search and updates it with the latest value (which is more consistent with search input value).
> 
> Hmm ok, so you are actually requesting a search. Fine for now, remember it should not be necessary if the component remains

The `SearchRouteReuseStrategy` actually seems to work. When navigating back from "Organisations" to "Search", the displayed images are not fetched, for instance. The results are reset, however, so I need to call `updateSearch`. This is what we do when navigating to a md record and back as well, where we have/had the same behavior.

**Comment by fgravin** [comment]
Oh yes, I think it's to set the router state back again (meaning we have the filters in the url back)

**Comment by fgravin** [comment]
Ok I have detached all home routes for the datahub which fixes the scroll issue.
Nevertheless, I think we gonna have an issue when news and organisations page will have a scroll too, it will trigger the search.

Will fix that later on.
@tkohr if it's ready on your side you can merge, thanks

**Comment by tkohr** [comment]
Great, thanks for the fix @fgravin, both `infiniteScroll` and `SearchRouteReuseStrategy` work this way for now :+1:

**Pull request merged**